vault-tool-server/src/Network/VaultTool/VaultServerProcess.hs

```diff
@@ -101,8 +102,7 @@ readVaultBackendConfig file = do
 -- | File should have one line per key (blank lines are ignored)
 readVaultUnsealKeys :: FilePath -> IO [VaultUnsealKey]
 readVaultUnsealKeys file =
-    T.readFile file >>=
-        (pure . map VaultUnsealKey . (filter (not . T.null)) . map T.strip . T.lines)
+    T.readFile file <&> (map VaultUnsealKey . filter (not . T.null) . map T.strip . T.lines)
```

Comment:
Good catch. hlint suggested `<&>` and I made that change without thinking too much about it.

Reply:
> Since we are already changing the API a bit (major version change), let's talk about making some additional changes.
> 
> Would it be possible to refactor so that:
> 
> * API functions that make unauthenticated requests take a `Manager` and `VaultAddress` (I'd like to reuse connections if possible)
> * Get rid of the `UnauthenticatedVaultConnection` constructor
> * API functions that make authenticated requests take a `VaultConnection` rather than calling `newTlsManager` (getting here might be a no-op)

If I understand this correctly, this refactor applies to the code in `Network.VaultTool`? The code in `Network.VaultTool.KeyValueV2` is already designed to take a `VaultConnection` (and doesn't have any unauthenticated calls). So apply the same approach to `Network.VaultTool`?